### PR TITLE
Add IPAM, VNV, update Security Admin and Vnet specs

### PR DESCRIPTION
**Pull request by sssharma24**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Comment by sssharma24** [comment]
All the changes are from these 2 PRs which have been reviewed and approved by ARM reviewer:
https://github.com/Azure/azure-rest-api-specs-pr/pull/19000
https://github.com/Azure/azure-rest-api-specs-pr/pull/18922

File: specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/ipAddressManager.json [hunk 1]
```diff
@@ -0,0 +1,1119 @@
...
+          "201": {
+            "description": "Pool was created",
+            "schema": {
+              "$ref": "#/definitions/IpamPool"
+            },
+            "headers": {
+              "Azure-AsyncOperation": {
+                "type": "string"
+              }
+            }
```
Comment (razvanbadea-msft) [inline specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/ipAddressManager.json:150]: compared to 18922 from private this was added only in this pr

File: specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/ipAddressManager.json [hunk 2]
```diff
@@ -0,0 +1,1119 @@
...
+            "description": "Default error response.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CommonErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        },
```
Comment (razvanbadea-msft) [inline specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/ipAddressManager.json:162]: same here

File: specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/ipAddressManager.json [hunk 3]
```diff
@@ -0,0 +1,1119 @@
...
+            "pattern": "^[a-zA-Z0-9-]*$"
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Pool resource object to create/update.",
+            "required": true,
```
Comment (razvanbadea-msft) [inline specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/ipAddressManager.json:128]: >"required": true, [](http://example.com/codeflow?start=12&length=17)

compared to 18922 from private this was added only in this pr

File: specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/ipAddressManager.json [hunk 4]
```diff
@@ -0,0 +1,1119 @@
...
+        "modelAsString": true,
+        "name": "IpType"
+      }
+    },
+    "IpamPool": {
+      "description": "Instance of Pool resource.",
+      "type": "object",
+      "required": [
+        "properties"
+      ],
```
Comment (razvanbadea-msft) [inline specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/ipAddressManager.json:766]: compared to 18922 from private this was added only in this pr

File: specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/networkVerifier.json [hunk 5]
```diff
@@ -0,0 +1,1247 @@
...
+          "description": "Gets the URL to get the next page of results.",
+          "type": "string"
+        }
+      },
+      "description": "A list of Verifier Workspace"
+    },
+    "VerifierWorkspace": {
+      "description": "Instance of Verifier Workspace.",
+      "type": "object",
+      "properties": {
```
Comment (razvanbadea-msft) [inline specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/networkVerifier.json:1176]: >properties [](http://example.com/codeflow?start=7&length=10)

compared with the private version this one is missing the 

"location": {
                    "type": "string",
                    "x-ms-mutability": [
                        "create",
                        "read"
                    ]
                },

File: specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/networkVerifier.json [hunk 6]
```diff
@@ -0,0 +1,1247 @@
...
+            "in": "path",
+            "name": "workspaceName",
+            "description": "Workspace name.",
+            "required": true,
+            "type": "string",
+            "pattern": "^[a-zA-Z0-9_.-]*$"
+          }
+        ],
+        "responses": {
+          "202": {
```
Comment (razvanbadea-msft) [inline specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/networkVerifier.json:881]: >202 [](http://example.com/codeflow?start=11&length=3)

same here 

"headers": {
                            "Location": {
                                "description": "The URL of the resource used to check the status of the asynchronous operation.",
                                "type": "string"
                            }
                        }

File: specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/networkVerifier.json [hunk 7]
```diff
@@ -0,0 +1,1247 @@
...
+            "description": "VerifierWorkspace delete request succeeded."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CommonErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
```
Comment (razvanbadea-msft) [inline specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/networkVerifier.json:894]: >"x-ms-long-running-operation": true, [](http://example.com/codeflow?start=8&length=36)

also here 

"x-ms-long-running-operation-options": {
                    "final-state-via": "location"
                }

File: specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/networkVerifier.json [hunk 8]
```diff
@@ -0,0 +1,1247 @@
...
+            "in": "path",
+            "name": "reachabilityAnalysisRunName",
+            "description": "Reachability Analysis Run name.",
+            "required": true,
+            "type": "string",
+            "pattern": "^[a-zA-Z0-9_.-]*$"
+          }
+        ],
+        "responses": {
+          "202": {
```
Comment (razvanbadea-msft) [inline specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/networkVerifier.json:563]: >202 [](http://example.com/codeflow?start=11&length=3)

missing the

File: specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/networkVerifier.json [hunk 9]
```diff
@@ -0,0 +1,1247 @@
...
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CommonErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
```
Comment (razvanbadea-msft) [inline specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/networkVerifier.json:516]: >"x-ms-long-running-operation": true, [](http://example.com/codeflow?start=8&length=36)

same here 

"x-ms-long-running-operation-options": {
                    "final-state-via": "location"
                }

File: specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/ipAddressManager.json (hunk 1 above)
Comment (arjun-d-patel) [reply specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/ipAddressManager.json:150]: Yes, this was added for ipAddressManager after the private PR. This added to support long running PUTs for IpamPools resource.

File: specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/ipAddressManager.json (hunk 2 above)
Comment (arjun-d-patel) [reply specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/ipAddressManager.json:162]: Yes same as here: https://github.com/Azure/azure-rest-api-specs/pull/30560#discussion_r1764051917

File: specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/ipAddressManager.json (hunk 3 above)
Comment (arjun-d-patel) [reply specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/ipAddressManager.json:128]: We added this for azure client tools like CLI to generate the right required parameters. Yes, this was after the private PR.

File: specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/ipAddressManager.json (hunk 4 above)
Comment (arjun-d-patel) [reply specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/ipAddressManager.json:766]: Yes same as here: https://github.com/Azure/azure-rest-api-specs/pull/30560#discussion_r1764053823

File: specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/networkVerifier.json (hunk 8 above)
Comment (sssharma24) [reply specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/networkVerifier.json:562]: added back, we initially were not going to support location headers for analysisrun delete and verifierworkspace delete but we are only not supporting location headers in analysis run put anymore

File: specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/networkVerifier.json (hunk 6 above)
Comment (sssharma24) [reply specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/networkVerifier.json:889]: same as here https://github.com/Azure/azure-rest-api-specs/pull/30560/files#r1764369061

File: specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/networkVerifier.json (hunk 7 above)
Comment (sssharma24) [reply specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/networkVerifier.json:908]: same as here https://github.com/Azure/azure-rest-api-specs/pull/30560/files#r1764369061

File: specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/networkVerifier.json (hunk 5 above)
Comment (sssharma24) [reply specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/networkVerifier.json:1193]: removed those as they were duplicates and causing error in sdk generation for another team, already approved and merged here https://github.com/Azure/azure-rest-api-specs/pull/30327

File: specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/networkVerifier.json (hunk 9 above)
Comment (sssharma24) [reply specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/networkVerifier.json]: same as here https://github.com/Azure/azure-rest-api-specs/pull/30560/files#r1764369061

Comment (sssharma24) [reply specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/networkVerifier.json]: this is a new change not reviewed

**Comment by sssharma24** [comment]
> > ```
> >     "Routing"
> > ```
> 
> thsi values are newly added is it as expected?
> 
> Refers to: specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/networkManager.json:693 in [6b006e5](https://github.com/Azure/azure-rest-api-specs/commit/6b006e5dcb49cf56aedf0b9f8b55a3f1fcc7aca1). [](commit_id = [6b006e5](https://github.com/Azure/azure-rest-api-specs/commit/6b006e5dcb49cf56aedf0b9f8b55a3f1fcc7aca1), deletion_comment = False)

yes this have already been merged as well https://github.com/Azure/azure-rest-api-specs-pr/pull/18788/files

**Comment by sssharma24** [comment]
@razvanbadea-msft can I remove the breaking change for sdk go and js tags since both those tests are passing?

**Comment by razvanbadea-msft** [comment]
> @razvanbadea-msft can I remove the breaking change for sdk go and js tags since both those tests are passing?

no, these needs to be approved/re-approved, it will be reviewed in step 3

**Comment by qiaozha** [comment]
@tadelesh could help on the Go SDK breaking change.

**Comment by tadelesh** [comment]
in this pr, some base resource refs have been changed which cause go sdk breaking. @sssharma24 did you get approval from breaking change review board?

**Comment by sssharma24** [comment]
@tadelesh the sdk go check actually passed and these changes have already been approved for preview versions as well. this is mostly just merging it to the stable

**Comment by tadelesh** [comment]
could you 

> All the changes are from these 2 PRs which have been reviewed and approved by ARM reviewer: [Azure/azure-rest-api-specs-pr#19000](https://github.com/Azure/azure-rest-api-specs-pr/pull/19000) [Azure/azure-rest-api-specs-pr#18922](https://github.com/Azure/azure-rest-api-specs-pr/pull/18922)

actually, i did not see the breaking change review in the private pr

**Comment by sssharma24** [comment]
> could you
> 
> > All the changes are from these 2 PRs which have been reviewed and approved by ARM reviewer: [Azure/azure-rest-api-specs-pr#19000](https://github.com/Azure/azure-rest-api-specs-pr/pull/19000) [Azure/azure-rest-api-specs-pr#18922](https://github.com/Azure/azure-rest-api-specs-pr/pull/18922)
> 
> actually, i did not see the breaking change review in the private pr

this is the main PR in priv repo with most changes and the sdk .net failed here as well but we were given the Ok to merged tag https://github.com/Azure/azure-rest-api-specs-pr/pull/17956

**Comment by tadelesh** [comment]
> > could you
> > > All the changes are from these 2 PRs which have been reviewed and approved by ARM reviewer: [Azure/azure-rest-api-specs-pr#19000](https://github.com/Azure/azure-rest-api-specs-pr/pull/19000) [Azure/azure-rest-api-specs-pr#18922](https://github.com/Azure/azure-rest-api-specs-pr/pull/18922)
> > 
> > 
> > actually, i did not see the breaking change review in the private pr
> 
> this is the main PR in priv repo with most changes and the breaking change benign tag #29998

thanks for the info, i approved go breaking changes.

**Comment by MUHAMMADIHSANALIKAP** [comment]
Ok.
@MUHAMMAD IHSAN ALI.

**Pull request merged**